src/core/namespace.c

```diff
@@ -598,11 +598,24 @@ static int clone_device_node(const char *d, const char *temporary_mount) {
 
         dn = strjoina(temporary_mount, d);
 
-        mac_selinux_create_file_prepare(d, st.st_mode);
-        r = mknod(dn, st.st_mode, st.st_rdev);
-        mac_selinux_create_file_clear();
+        if (do_mknod) {
+                mac_selinux_create_file_prepare(d, st.st_mode);
+                r = mknod(dn, st.st_mode, st.st_rdev);
+                mac_selinux_create_file_clear();
+        } else {
+                _cleanup_close_ int fd = -1;
+
+                mac_selinux_create_file_prepare(d, st.st_mode);
+                fd = open(dn, O_CREAT | O_EXCL | O_CLOEXEC, st.st_mode);
+                mac_selinux_create_file_clear();
+                if (fd < 0) {
+                        if (errno != EEXIST)
+                                return -errno;
+                }
+                r = mount(d, dn, NULL, MS_BIND, NULL);
```

Comment:
So I am normally very much concerned about bind mounting device nodes into something that might use RootDirectory=/RootImage=, since device nodes carry file ownership/ACLs and such, and UID/GID tables generally don't match between such payload and host environments. That said, the listed device nodes are either owned by root:root (where this is a non-issue) or by root:tty (in the case of /dev/tty and /dev/ptmx), and generally do not carry ACLs. Now, "tty" is actually kinda special, since it's the one GID we compile into systemd, since we need to resolve it very very early on before user/group name resolution is available. When we decided to do that we did some homework figuring out the different values that distros set this GID to, and as it turns out they all agree on 5 for this one (yay), hence I figure we can get away with bind mounting these two nodes too.

Hence, I figure this is OK to go in, but this really really needs some comment here, explaining that the assumption here is that everybody agrees that gid(tty)==5, and that it will break if that's not true, and that the assumption here is that root:tty and root:root are the only owners of these specific device nodes processed here...

Reply:
Cool, I'll do the changes soonish. If all goes well I can do it on a flight I have to take on Sunday. That'll be the optimal scenario. If not this would maybe need to sit for ~1 week from said Sunday on. That ok?

Reply:
yes, absolutely. Thanks for looking into this!

Reply:
addressed